plugin/parser/differ/mysql/differ.go

```diff
@@ -0,0 +1,45 @@
+// Package mysql provides the MySQL differ plugin.
+package mysql
+
+import (
+	"bytes"
+
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/parser/format"
+)
+
+// SchemaDiff returns the schema diff.
+func SchemaDiff(old, new []ast.StmtNode) (string, error) {
+	oldTableList := make([]*ast.CreateTableStmt, 0)
+	oldTableMap := make(map[string]int)
+	tableDiff := make([]ast.Node, 0)
+	for _, node := range old {
+		if stmt, ok := node.(*ast.CreateTableStmt); ok {
+			tableName := stmt.Table.Name.String()
+			oldTableMap[tableName] = len(oldTableList)
```

Comment:
Why do we need to maintain the `len(oldTableList)`

Reply:
The map in golang is unordered. We may need to deal with reference order, so the slice is more suitable for us.
Now we need to find out which tables need to be created. Adding a map to check for existence is to reduce the complexity from O(mn) to O(n).

Reply:
But you're not actually using it.

Conversely, the creation order is determined by table order in `new`, instead of the `old`.

Reply:
Oh... SGTM. We don't need to care about the order of the `old` SDL.